pre-commit.sh

```diff
@@ -11,9 +11,9 @@ fi
 venv/bin/pip install -e .
 
 # Generate docs
-venv/bin/python scripts/gen-rest-docs.py || exit 1  # Outputs to `docs`
-venv/bin/python scripts/gen-cli-docs.py || exit 1  # Outputs to `docs`
-venv/bin/mkdocs build || exit 1  # Outputs to `site`
+venv/bin/python scripts/gen-rest-docs.py || { rm -rf venv; exit 1; }  # Outputs to `docs`
```

Comment:
Why do we want to delete the venv folder on failure? Failure of `scripts/gen-rest-docs.py` doesn't necessarily mean that something was wrong with the venv -- I could imagine that automatic deletion of the env could serve as a cause of frustration for many, and it's not too difficult to do yourself.

Reply:
I saw that venv was corrupted (I'm not sure what about it was deleting it mitigated the issues I saw). I think removing it is a reasonable thing to do since it fixes the issue I was hitting on a clean ubuntu vm but I can remove this step if you'd like. FWIW, I thought I had merged this pr but seems I didn't.